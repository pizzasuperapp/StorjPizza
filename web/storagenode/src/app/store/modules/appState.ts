// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

export const APPSTATE_MUTATIONS = {
    TOGGLE_SATELLITE_SELECTION: 'TOGGLE_SATELLITE_SELECTION',
    TOGGLE_BANDWIDTH_CHART: 'TOGGLE_BANDWIDTH_CHART',
    TOGGLE_EGRESS_CHART: 'TOGGLE_EGRESS_CHART',
    TOGGLE_INGRESS_CHART: 'TOGGLE_INGRESS_CHART,',
    TOGGLE_PAYOUT_CALENDAR: 'TOGGLE_PAYOUT_CALENDAR',
    CLOSE_ADDITIONAL_CHARTS: 'CLOSE_ADDITIONAL_CHARTS',
    CLOSE_ALL_POPUPS: 'CLOSE_ALL_POPUPS',
    SET_DARK: 'SET_DARK',
    SET_NO_PAYOUT_INFO: 'SET_NO_PAYOUT_INFO',
    SET_LOADING_STATE: 'SET_LOADING_STATE',
    TOGGLE_PAYOUT_HISTORY_CALENDAR: 'TOGGLE_PAYOUT_HISTORY_CALENDAR',
};

export const APPSTATE_ACTIONS = {
    TOGGLE_SATELLITE_SELECTION: 'TOGGLE_SATELLITE_SELECTION',
    TOGGLE_BANDWIDTH_CHART: 'TOGGLE_BANDWIDTH_CHART',
    TOGGLE_EGRESS_CHART: 'TOGGLE_EGRESS_CHART',
    TOGGLE_INGRESS_CHART: 'TOGGLE_INGRESS_CHART',
    CLOSE_ADDITIONAL_CHARTS: 'CLOSE_ADDITIONAL_CHARTS',
    TOGGLE_PAYOUT_CALENDAR: 'TOGGLE_PAYOUT_CALENDAR',
    CLOSE_ALL_POPUPS: 'CLOSE_ALL_POPUPS',
    SET_DARK_MODE: 'SET_DARK_MODE',
    SET_NO_PAYOUT_DATA: 'SET_NO_PAYOUT_DATA',
    SET_LOADING: 'SET_LOADING',
    TOGGLE_PAYOUT_HISTORY_CALENDAR: 'TOGGLE_PAYOUT_HISTORY_CALENDAR',
};

const {
    TOGGLE_SATELLITE_SELECTION,
    TOGGLE_BANDWIDTH_CHART,
    TOGGLE_EGRESS_CHART,
    TOGGLE_INGRESS_CHART,
    CLOSE_ADDITIONAL_CHARTS,
    CLOSE_ALL_POPUPS,
    TOGGLE_PAYOUT_CALENDAR,
} = APPSTATE_MUTATIONS;

class AppState {
    public constructor (
        public isSatelliteSelectionShown = false,
        public isBandwidthChartShown = true,
        public isEgressChartShown = false,
        public isIngressChartShown = false,
        public isPayoutCalendarShown = false,
        public isDarkMode = false,
        public isNoPayoutData = false,
        public isLoading = true,
        public isPayoutHistoryCalendarShown = false,
    ) {}
}

interface AppContext {
    state: AppState;
    commit: (string, ...unknown) => void;
}

export const appStateModule = {
    state: new AppState(),
    mutations: {
        [TOGGLE_SATELLITE_SELECTION](state: AppState): void {
            state.isSatelliteSelectionShown = !state.isSatelliteSelectionShown;
        },
        [TOGGLE_BANDWIDTH_CHART](state: AppState): void {
            state.isBandwidthChartShown = !state.isBandwidthChartShown;
        },
        [TOGGLE_EGRESS_CHART](state: AppState): void {
            state.isEgressChartShown = !state.isEgressChartShown;
        },
        [TOGGLE_INGRESS_CHART](state: AppState): void {
            state.isIngressChartShown = !state.isIngressChartShown;
        },
        [TOGGLE_PAYOUT_CALENDAR](state: AppState, value: boolean): void {
            state.isPayoutCalendarShown = value;
        },
        [CLOSE_ADDITIONAL_CHARTS](state: AppState): void {
            state.isBandwidthChartShown = true;
            state.isIngressChartShown = false;
            state.isEgressChartShown = false;
        },
        [APPSTATE_MUTATIONS.SET_DARK](state: AppState, value: boolean): void {
            state.isDarkMode = value;
        },
        [APPSTATE_MUTATIONS.SET_NO_PAYOUT_INFO](state: AppState, value: boolean): void {
            state.isNoPayoutData = value;
        },
        [APPSTATE_MUTATIONS.SET_LOADING_STATE](state: AppState, value: boolean): void {
            state.isLoading = value;
        },
        [CLOSE_ALL_POPUPS](state: AppState): void {
            state.isSatelliteSelectionShown = false;
        },
        [APPSTATE_MUTATIONS.TOGGLE_PAYOUT_HISTORY_CALENDAR](state: AppState, value: boolean): void {
            state.isPayoutHistoryCalendarShown = value;
        },
    },
    actions: {
        [APPSTATE_ACTIONS.TOGGLE_SATELLITE_SELECTION]: function ({commit, state}: AppContext): void {
            if (!state.isSatelliteSelectionShown) {
                commit(APPSTATE_MUTATIONS.TOGGLE_SATELLITE_SELECTION);

                return;
            }

            commit(APPSTATE_MUTATIONS.CLOSE_ALL_POPUPS);
        },
        [APPSTATE_ACTIONS.TOGGLE_PAYOUT_CALENDAR]: function ({commit}: AppContext, value: boolean): void {
            commit(APPSTATE_MUTATIONS.TOGGLE_PAYOUT_CALENDAR, value);
        },
        [APPSTATE_ACTIONS.TOGGLE_EGRESS_CHART]: function ({commit, state}: AppContext): void {
            if (!state.isBandwidthChartShown) {
                commit(APPSTATE_MUTATIONS.TOGGLE_EGRESS_CHART);
                commit(APPSTATE_MUTATIONS.TOGGLE_INGRESS_CHART);

                return;
            }

            commit(APPSTATE_MUTATIONS.TOGGLE_BANDWIDTH_CHART);
            commit(APPSTATE_MUTATIONS.TOGGLE_EGRESS_CHART);
        },
        [APPSTATE_ACTIONS.TOGGLE_INGRESS_CHART]: function ({commit, state}: AppContext): void {
            if (!state.isBandwidthChartShown) {
                commit(APPSTATE_MUTATIONS.TOGGLE_INGRESS_CHART);
                commit(APPSTATE_MUTATIONS.TOGGLE_EGRESS_CHART);

                return;
            }

            commit(APPSTATE_MUTATIONS.TOGGLE_BANDWIDTH_CHART);
            commit(APPSTATE_MUTATIONS.TOGGLE_INGRESS_CHART);
        },
        [APPSTATE_ACTIONS.SET_DARK_MODE]: function ({commit}: AppContext, value: boolean): void {
            commit(APPSTATE_MUTATIONS.SET_DARK, value);
        },
        [APPSTATE_ACTIONS.SET_NO_PAYOUT_DATA]: function ({commit}: AppContext, value: boolean): void {
            commit(APPSTATE_MUTATIONS.SET_NO_PAYOUT_INFO, value);
        },
        [APPSTATE_ACTIONS.SET_LOADING]: function ({commit}: AppContext, value: boolean): void {
            value ? commit(APPSTATE_MUTATIONS.SET_LOADING_STATE, value) :
                setTimeout(() => { commit(APPSTATE_MUTATIONS.SET_LOADING_STATE, value); }, 1000);
        },
        [APPSTATE_ACTIONS.CLOSE_ADDITIONAL_CHARTS]: function ({commit}: AppContext): void {
            commit(APPSTATE_MUTATIONS.CLOSE_ADDITIONAL_CHARTS);
        },
        [APPSTATE_ACTIONS.CLOSE_ALL_POPUPS]: function ({commit}: AppContext): void {
            commit(APPSTATE_MUTATIONS.CLOSE_ALL_POPUPS);
        },
        [APPSTATE_ACTIONS.TOGGLE_PAYOUT_HISTORY_CALENDAR]: function ({commit}: AppContext, value: boolean): void {
            commit(APPSTATE_MUTATIONS.TOGGLE_PAYOUT_HISTORY_CALENDAR, value);
        },
    },
};
