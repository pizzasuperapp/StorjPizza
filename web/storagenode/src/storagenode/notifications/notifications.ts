// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * Exposes all notifications-related functionality.
 */
export interface NotificationsApi {
    /**
     * Fetches notifications.
     * @throws Error
     */
    get(cursor: NotificationsCursor): Promise<NotificationsResponse>;

    /**
     * Marks single notification as read.
     * @throws Error
     */
    read(id: string): Promise<void>;

    /**
     * Marks all notification as read.
     * @throws Error
     */
    readAll(): Promise<void>;
}

/**
 * Describes notification entity.
 */
export class Notification {
    public constructor(
        public id: string = '',
        public senderId: string = '',
        public type: NotificationTypes = NotificationTypes.Custom,
        public title: string = '',
        public message: string = '',
        public readAt: Date | null = null,
        public createdAt: Date = new Date(),
    ) {}
}

/**
 * Describes all current notifications types.
 */
export enum NotificationTypes {
    Custom = 0,
    AuditCheckFailure = 1,
    Disqualification = 2,
    Suspension = 3,
}

/**
 * Describes page offset for pagination.
 */
export class NotificationsCursor {
    private DEFAULT_LIMIT = 7;

    public constructor(
        public page: number = 0,
        public limit: number = 0,
    ) {
        if (!this.limit) {
            this.limit = this.DEFAULT_LIMIT;
        }
    }
}

/**
 * Describes response object from server.
 */
export class NotificationsResponse {
    public constructor(
        public page: NotificationsPage = new NotificationsPage(),
        public unreadCount: number = 0,
        public totalCount: number = 0,
    ) {}
}

/**
 * Describes page related notification information.
 */
export class NotificationsPage {
    public constructor(
        public notifications: Notification[] = [],
        public pageCount: number = 0,
    ) {}
}
