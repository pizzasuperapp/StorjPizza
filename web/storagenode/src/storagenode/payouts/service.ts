// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import {
    EstimatedPayout,
    PaymentInfoParameters,
    PayoutApi,
    PayoutPeriod,
    Paystub,
    SatelliteHeldHistory,
    SatellitePayoutForPeriod,
    TotalPayments,
    TotalPaystubForPeriod,
} from '@/storagenode/payouts/payouts';

/**
 * PayoutService is used to store and handle node paystub information.
 * PayoutService exposes a business logic related to payouts.
 */
export class PayoutService {
    private readonly payouts: PayoutApi;

    public constructor(api: PayoutApi) {
        this.payouts = api;
    }

    /**
     * Gets summary of paystubs for given period.
     * @param start period start
     * @param end period end
     * @param satelliteId
     */
    public async paystubSummaryForPeriod(start: PayoutPeriod | null, end: PayoutPeriod, satelliteId?: string): Promise<TotalPaystubForPeriod> {
        const paystubs: Paystub[] = await this.payouts.getPaystubsForPeriod(new PaymentInfoParameters(start, end, satelliteId));

        return new TotalPaystubForPeriod(paystubs);
    }

    /**
     * Gets held and paid summary for given period.
     * @param start period start
     * @param end period end
     * @param satelliteId
     */
    public async totalPayments(start: PayoutPeriod, end: PayoutPeriod, satelliteId: string): Promise<TotalPayments> {
        const paystubs: Paystub[] = await this.payouts.getPaystubsForPeriod(new PaymentInfoParameters(start, end, satelliteId));

        return new TotalPayments(paystubs);
    }

    /**
     * Gets list of payout periods that have paystubs for selected satellite.
     * If satelliteId is not provided returns periods for all satellites.
     * @param satelliteId
     */
    public async availablePeriods(satelliteId: string): Promise<PayoutPeriod[]> {
        return await this.payouts.getPayoutPeriods(satelliteId);
    }

    /**
     * Gets list of payout history items for given period by satellites.
     * @param payoutHistoryPeriod year and month representation
     */
    public async payoutHistory(payoutHistoryPeriod: string): Promise<SatellitePayoutForPeriod[]> {
        return await this.payouts.getPayoutHistory(payoutHistoryPeriod);
    }

    /**
     * Gets list of held history for all satellites.
     */
    public async allSatellitesHeldHistory(): Promise<SatelliteHeldHistory[]> {
        return await this.payouts.getHeldHistory();
    }

    /**
     * Gets estimated payout when no data in paystub.
     * @param satelliteId
     */
    public async estimatedPayout(satelliteId: string): Promise<EstimatedPayout> {
        return await this.payouts.getEstimatedPayout(satelliteId);
    }
}
