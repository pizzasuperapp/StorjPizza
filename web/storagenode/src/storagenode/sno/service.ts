// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import { StorageNodeApi } from '@/storagenode/api/storagenode';
import { Dashboard, Satellite, Satellites } from '@/storagenode/sno/sno';

/**
 * SNOService is used to store and handle node information.
 * SNOService exposes a business logic related to node.
 */
export class StorageNodeService {
    private readonly node: StorageNodeApi;

    public constructor(api: StorageNodeApi) {
        this.node = api;
    }

    /**
     * Gets dashboard data from server.
     */
    public async dashboard(): Promise<Dashboard> {
        return await this.node.dashboard();
    }

    /**
     * Gets satellite data from server.
     * @param id - satellite id
     */
    public async satellite(id: string): Promise<Satellite> {
        return await this.node.satellite(id);
    }

    /**
     * Gets all satellites data from server.
     */
    public async satellites(): Promise<Satellites> {
        return await this.node.satellites();
    }
}
