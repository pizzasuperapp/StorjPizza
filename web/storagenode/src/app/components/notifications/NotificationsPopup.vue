// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="notification-popup-container">
        <div class="notification-popup-container__header">
            <p class="notification-popup-container__header__title">Notifications</p>
            <a :href="notificationsPath" class="notification-popup-container__header__link">
                <p>See All</p>
            </a>
        </div>
        <div
            v-if="latest.length"
            class="notification-popup-container__content"
            :class="{'collapsed': isCollapsed}"
        >
            <SNONotification
                v-for="notification in latest"
                :key="notification.id"
                is-small="true"
                :notification="notification"
            />
        </div>
        <div v-else class="notification-popup-container__empty-state">
            <img src="@/../static/images/notifications/EmptyState.png" alt="Empty state image">
            <p class="notification-popup-container__empty-state__label">No notifications yet</p>
        </div>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import SNONotification from '@/app/components/notifications/SNONotification.vue';

import { RouteConfig } from '@/app/router';
import { UINotification } from '@/app/types/notifications';

// @vue/component
@Component({
    components: {
        SNONotification,
    },
})
export default class NotificationsPopup extends Vue {
    /**
     * Path to notifications route.
     */
    public readonly notificationsPath: string = RouteConfig.Notifications.path;

    /**
     * Represents first page of notifications.
     */
    public get latest(): UINotification[] {
        return this.$store.state.notificationsModule.latestNotifications;
    }

    /**
     * Indicates if popup is smaller than with scroll.
     */
    public get isCollapsed(): boolean {
        return this.latest.length < 4;
    }
}
</script>

<style scoped lang="scss">
    .notification-popup-container {
        position: relative;
        width: 400px;
        height: auto;
        max-height: 376px;
        background-color: var(--block-background-color);
        border-radius: 12px;
        padding: 27px 0 10px;
        box-shadow: 0 7px 17px var(--block-background-color);
        z-index: 104;

        &__header {
            display: flex;
            justify-content: space-between;
            align-items: center;

            &__title {
                font-family: 'font_bold', sans-serif;
                font-size: 24px;
                line-height: 36px;
                color: var(--title-text-color);
                margin-left: 32px;
            }

            &__link {
                font-family: 'font_regular', sans-serif;
                font-size: 14px;
                color: var(--navigation-link-color);
                margin-right: 20px;
                text-decoration: none;
            }
        }

        &__content {
            height: 300px;
            overflow-y: scroll;
        }

        &__empty-state {
            width: 100%;
            display: flex;
            flex-direction: column;
            align-items: center;
            justify-content: center;
            padding: 46px 0;

            &__label {
                margin-top: 35px;
                font-family: 'font_regular', sans-serif;
                font-size: 16px;
                color: var(--regular-text-color);
            }
        }
    }

    .collapsed {
        height: auto !important;
    }

    @media screen and (max-width: 460px) {

        .notification-popup-container {
            width: 100%;
            max-height: 350px;
        }
    }
</style>
