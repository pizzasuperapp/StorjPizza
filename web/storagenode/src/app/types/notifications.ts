// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import Vue, {VueConstructor} from 'vue';

import { Notification, NotificationTypes } from '@/storagenode/notifications/notifications';

import DisqualificationIcon from "@/../static/images/notifications/disqualified.svg";
import FailIcon from "@/../static/images/notifications/fail.svg";
import InfoIcon from "@/../static/images/notifications/info.svg";
import SuspendedIcon from "@/../static/images/notifications/suspended.svg";

/**
 * Holds all notifications module state.
 */
export class NotificationsState {
    public latestNotifications: UINotification[] = [];

    public constructor(
        public notifications: UINotification[] = [],
        public pageCount: number = 0,
        public unreadCount: number = 0,
    ) { }
}

/**
 * Describes notification entity.
 */
export class UINotification {
    public isRead: boolean;
    public id: string;
    public senderId: string;
    public type: NotificationTypes;
    public title: string;
    public message: string;
    public readAt: Date | null;
    public createdAt: Date = new Date();

    public constructor(notification: Partial<UINotification> = new Notification()) {
        Object.assign(this, notification);
        this.isRead = !!this.readAt;
    }

    /**
     * dateLabels formats createdAt into more informative strings.
     */
    public get dateLabel(): string {
        const differenceInSeconds = (Math.trunc(new Date().getTime()) - Math.trunc(new Date(this.createdAt).getTime())) / 1000;

        switch (true) {
        case differenceInSeconds < 60:
            return 'Just now';
        case differenceInSeconds < 3600:
            return `${(differenceInSeconds / 60).toFixed(0)} minute(s) ago`;
        case differenceInSeconds < 86400:
            return `${(differenceInSeconds / 3600).toFixed(0)} hour(s) ago`;
        case differenceInSeconds < 86400 * 2:
            return `Yesterday`;
        default:
            return new Date(this.createdAt).toDateString();
        }
    }

    /**
     * markAsRead mark notification as read on UI.
     */
    public markAsRead(): void {
        this.isRead = true;
    }

    /**
     * setIcon selects notification icon depends on type.
     */
    public get icon(): VueConstructor<Vue> {
        switch (this.type) {
        case NotificationTypes.AuditCheckFailure:
            return FailIcon;
        case NotificationTypes.Disqualification:
            return DisqualificationIcon;
        case NotificationTypes.Suspension:
            return SuspendedIcon;
        default:
            return InfoIcon;
        }
    }
}
