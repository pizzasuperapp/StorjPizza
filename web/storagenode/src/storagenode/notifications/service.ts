// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import { NotificationsApi, NotificationsCursor, NotificationsResponse } from '@/storagenode/notifications/notifications';

/**
 * PayoutService is used to store and handle node paystub information.
 * PayoutService exposes a business logic related to payouts.
 */
export class NotificationsService {
    private readonly api: NotificationsApi;

    public constructor(api: NotificationsApi) {
        this.api = api;
    }

    /**
     * Fetch notifications.
     *
     * @returns notifications response.
     * @throws Error
     */
    public async notifications(index: number, limit?: number): Promise<NotificationsResponse> {
        const cursor = new NotificationsCursor(index, limit);

        return await this.api.get(cursor);
    }

    /**
     * Marks single notification as read on server.
     * @param id
     */
    public async readSingeNotification(id: string): Promise<void> {
        await this.api.read(id);
    }

    /**
     * Marks all notifications as read on server.
     */
    public async readAllNotifications(): Promise<void> {
        await this.api.readAll();
    }
}
