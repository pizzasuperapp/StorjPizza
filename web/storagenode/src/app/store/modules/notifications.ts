// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { StoreModule } from '@/app/store';
import { NotificationsState, UINotification } from '@/app/types/notifications';
import { NotificationsService } from '@/storagenode/notifications/service';

export const NOTIFICATIONS_MUTATIONS = {
    SET_NOTIFICATIONS: 'SET_NOTIFICATIONS',
    MARK_AS_READ: 'MARK_AS_READ',
    READ_ALL: 'READ_ALL',
    SET_LATEST: 'SET_LATEST',
};

export const NOTIFICATIONS_ACTIONS = {
    GET_NOTIFICATIONS: 'getNotifications',
    MARK_AS_READ: 'markAsRead',
    READ_ALL: 'readAll',
};

interface NotificationsContext {
    state: NotificationsState;
    commit: (string, ...unknown) => void;
}

/**
 * creates notifications module with all dependencies
 *
 * @param service - payments service
 */
export function newNotificationsModule(service: NotificationsService): StoreModule<NotificationsState> {
    return {
        state: new NotificationsState(),
        mutations: {
            [NOTIFICATIONS_MUTATIONS.SET_NOTIFICATIONS](state: NotificationsState, notificationsState: NotificationsState): void {
                state.notifications = notificationsState.notifications;
                state.pageCount = notificationsState.pageCount;
                state.unreadCount = notificationsState.unreadCount;
            },
            [NOTIFICATIONS_MUTATIONS.SET_LATEST](state: NotificationsState, notificationsState: NotificationsState): void {
                state.latestNotifications = notificationsState.notifications;
            },
            [NOTIFICATIONS_MUTATIONS.MARK_AS_READ](state: NotificationsState, id: string): void {
                state.notifications = state.notifications.map((notification: UINotification) => {
                    if (notification.id === id) {
                        notification.markAsRead();
                    }

                    return notification;
                });
            },
            [NOTIFICATIONS_MUTATIONS.READ_ALL](state: NotificationsState): void {
                state.notifications = state.notifications.map((notification: UINotification) => {
                    notification.markAsRead();

                    return notification;
                });

                state.unreadCount = 0;
            },
        },
        actions: {
            [NOTIFICATIONS_ACTIONS.GET_NOTIFICATIONS]: async function ({commit}: NotificationsContext, pageIndex: number): Promise<void> {
                const notificationsResponse = await service.notifications(pageIndex);

                const notifications = notificationsResponse.page.notifications.map(notification => new UINotification(notification));

                const notificationState = new NotificationsState(notifications, notificationsResponse.page.pageCount, notificationsResponse.unreadCount);

                commit(NOTIFICATIONS_MUTATIONS.SET_NOTIFICATIONS, notificationState);

                if (pageIndex === 1) {
                    commit(NOTIFICATIONS_MUTATIONS.SET_LATEST, notificationState);
                }
            },
            [NOTIFICATIONS_ACTIONS.MARK_AS_READ]: async function ({commit}: NotificationsContext, id: string): Promise<void> {
                await service.readSingeNotification(id);
                commit(NOTIFICATIONS_MUTATIONS.MARK_AS_READ, id);
            },
            [NOTIFICATIONS_ACTIONS.READ_ALL]: async function ({commit}: NotificationsContext): Promise<void> {
                await service.readAllNotifications();

                commit(NOTIFICATIONS_MUTATIONS.READ_ALL);
            },
        },
    };
}
