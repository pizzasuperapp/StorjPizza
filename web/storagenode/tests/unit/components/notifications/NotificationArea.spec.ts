// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import Router from 'vue-router';
import Vuex from 'vuex';

import { newNotificationsModule, NOTIFICATIONS_MUTATIONS } from '@/app/store/modules/notifications';
import { NotificationsState, UINotification } from '@/app/types/notifications';
import NotificationsArea from '@/app/views/NotificationsArea.vue';
import { NotificationsHttpApi } from '@/storagenode/api/notifications';
import { Notification } from '@/storagenode/notifications/notifications';
import { NotificationsService } from '@/storagenode/notifications/service';
import { createLocalVue, shallowMount } from '@vue/test-utils';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Router);

const notificationsApi = new NotificationsHttpApi();
const notificationsService = new NotificationsService(notificationsApi);
const notificationsModule = newNotificationsModule(notificationsService);

const store = new Vuex.Store({ modules: { notificationsModule }});

describe('NotificationsArea', (): void => {
    it('renders correctly with no notifications', (): void => {
        const wrapper = shallowMount(NotificationsArea, {
            store,
            localVue,
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with notifications', async (): Promise<void> => {
        await store.commit(NOTIFICATIONS_MUTATIONS.SET_NOTIFICATIONS, new NotificationsState(
            [new UINotification(new Notification())],
            1,
            1,
        ));

        const wrapper = shallowMount(NotificationsArea, {
            store,
            localVue,
        });

        expect(wrapper).toMatchSnapshot();
    });
});
