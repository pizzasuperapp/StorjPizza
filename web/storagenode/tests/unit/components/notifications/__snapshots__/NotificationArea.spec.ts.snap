// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsArea renders correctly with no notifications 1`] = `
<div class="notifications-container">
  <div class="notifications-container__header">
    <div class="notifications-container__header__left-area">
      <router-link-stub to="/" tag="a" ariacurrentvalue="page" event="click" class="notifications-container__header__back-link">
        <backarrowicon-stub></backarrowicon-stub>
      </router-link-stub>
      <p class="notifications-container__header__text">Notifications</p>
    </div> <button name="Mark all notifications as read" type="button" class="notifications-container__header__button disabled">
      <p class="notifications-container__header__button__label">Mark all as read</p>
    </button>
  </div>
  <div class="notifications-container__empty-state"><img src="@/../static/images/notifications/EmptyStateLarge.png" alt="Empty state image" class="notifications-container__empty-state__image">
    <p class="notifications-container__empty-state__label">No notifications yet</p>
  </div>
  <!---->
</div>
`;

exports[`NotificationsArea renders correctly with notifications 1`] = `
<div class="notifications-container">
  <div class="notifications-container__header">
    <div class="notifications-container__header__left-area">
      <router-link-stub to="/" tag="a" ariacurrentvalue="page" event="click" class="notifications-container__header__back-link">
        <backarrowicon-stub></backarrowicon-stub>
      </router-link-stub>
      <p class="notifications-container__header__text">Notifications</p>
    </div> <button name="Mark all notifications as read" type="button" class="notifications-container__header__button">
      <p class="notifications-container__header__button__label">Mark all as read</p>
    </button>
  </div>
  <div class="notifications-container__content-area">
    <snonotification-stub notification="[object Object]" class="notification"></snonotification-stub>
  </div>
  <!---->
</div>
`;
